### ISSUE TITLE
TopK Op hangs with Qwen Shapes and Specific Inputs

### ISSUE THREAD

**Issue by alingTT** [tenstorrent/tt-metal#32821]
With specific inputs from running the Qwen3-32B model, the topK op deterministically hangs on the first iteration of the following loop. I can forward the saved logits file to the person assigned this issue or the logits can be uploaded to some central location. If any information is missing please let me know.

To reproduce run the following unit test on BH-LB (8xp150) system.

```
import ttnn
import torch

def test_topk():
    mesh_device = ttnn.open_mesh_device(ttnn.MeshShape(1, 8))
    per_device_vocab_size = 18992
    indices_tensor_torch = torch.zeros(1, 1, 32, per_device_vocab_size, dtype=torch.int32)
    for i in range(per_device_vocab_size):
        indices_tensor_torch[:, :, :, i] = i
    tt_indices_tensor = ttnn.from_torch(
        indices_tensor_torch,
        dtype=ttnn.uint16,
        layout=ttnn.Layout.TILE,
        device=mesh_device,
        mesh_mapper=ttnn.ShardTensor2dMesh(mesh_device, dims=(None, None), mesh_shape=(1,8)),
        memory_config=ttnn.DRAM_MEMORY_CONFIG,
    )
    for i in range(100):
        print(f"Running topk {i}")
        torch_input = torch.load("topk_logits.pt") <- same logits saved from full Qwen model run
        tt_input = ttnn.from_torch(torch_input, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT,mesh_mapper=ttnn.ShardTensor2dMesh(mesh_device, dims=(None, 3), mesh_shape=(1,8)),memory_config=ttnn.L1_MEMORY_CONFIG, device=mesh_device)
        print("Running op")
        breakpoint()
        topk_values, topk_indices = ttnn.topk(tt_input, k=32, dim=-1, sub_core_grids=None, indices_tensor=tt_indices_tensor)
        print("Topk done")
        print(topk_values) # Hangs when printing output here
        print(topk_indices)

if __name__ == "__main__":
    test_topk()
```

**Comment by aczajkowskiTT** [comment]
@ejouretTT Per our conversation, the earliest I can begin to look at this is Friday.  
Extra comments from summary: There’s two parts to topk, one part involves bitonic sort and the latter that I implemented to support larger tensors and larger k uses insertion sort. Based on the above, this would involve the bitonic sort component, that has been heavily worked on by a few other folks I mentioned to you over the last few months.  Depending on the time within which you'd expect a resolution, we should discuss the best person to assign this to.

**Comment by alingTT** [comment]
@aczajkowskiTT 
Thanks for following up Adam, I attempted a fix here in this PR, I dont see any hangs for my specific shapes and model level output looks good. It would be great if you can review it when you have time to make sure that is a legitimate bug in the kernel. 

CC: @johanna-rock-tt

**Comment by aczajkowskiTT** [comment]
Resolved with this PR:
https://github.com/tenstorrent/tt-metal/pull/33287
